src/classes/AuthenticationSession.ts

```diff
@@ -32,6 +42,59 @@ export class AuthenticationSession {
     this.promptForPassword();
   }
 
+  public obtainNewAuthTokenUsingRefreshToken(): void {
+    if (!AuthenticationSession.sftpFusionAuthClientId) {
+      logger.error(
+        'Cannot obtain new access token without sftp client id.',
+      );
+      return;
+    }
+
+    if (!AuthenticationSession.sftpFusionAuthClientSecret) {
+      logger.error(
+        'Cannot obtain new access token without sftp client secret.',
+      );
+      return;
+    }
+
+    this.fusionAuthClient.exchangeRefreshTokenForAccessToken(
+      this.refreshToken,
+      AuthenticationSession.sftpFusionAuthClientId,
+      AuthenticationSession.sftpFusionAuthClientSecret,
+      '',
+      '',
+    )
+      .then((clientResponse) => {
```

Comment:
I believe this would be more readable with `async`/`await` than using the Promise API. You're using `async`/`await` elsewhere in this PR; is there a reason not to here?

Reply:
Not just an issue of readability -- the core functionality of the function is to generate and set an instance attribute.  By hiding the fact that it's async in the function signature we're creating room for race conditions.

This function *must* be marked as async!

Reply:
I don't know if my comment got lost in GitHub UI or I never made it but I just wanna underscore that as written this introduces a race condition and we *must* use async / await since the calling code relies on the result of this async call.